--- ./configure.ac.org	2006-05-25 13:51:44.000000000 +0000
+++ ./configure.ac	2007-07-14 11:58:16.000000000 +0000
@@ -128,7 +128,7 @@
 # Check for libcrypto using pkg-config, defaulting to /usr/lib
 AC_MSG_CHECKING(OpenSSL's crypto lib)
 CRYPTO_LIB_DIR="$(pkg-config openssl --variable=libdir)"
-CRYPTO_LIB="${CRYPTO_LIB_DIR:-/usr/lib}/libcrypto.a"
+CRYPTO_LIB="-lcrypto"
 AC_MSG_RESULT($CRYPTO_LIB)
 AC_SUBST(CRYPTO_LIB)
 
--- ./Makefile.org	2006-10-16 20:05:43.000000000 +0000
+++ ./Makefile	2007-07-14 12:25:41.000000000 +0000
@@ -75,14 +75,14 @@
 
 ifeq ($(OCAMLBEST), opt)
 monotone-viz : $(OBJX) lib3rdparty.a
-	$(OCAMLOPT) -o $@ -I $(LABLGTK_DIR) $(MLLIBS) $^ $(CRYPTO_LIB) $(SQLITE_LINK)
+	$(OCAMLOPT) -o $@ -I $(LABLGTK_DIR) $(MLLIBS) $^ -cclib "$(CRYPTO_LIB)" $(SQLITE_LINK)
 monotone-viz : MLLIBS = str.cmxa lablgtk.cmxa gtkInit.cmx lablgnomecanvas.cmxa 
 ifeq ($(LOCAL_LABLGTK), yes)
 $(OBJX) : .lablgtk
 endif
 else
 monotone-viz : $(OBJ) lib3rdparty.a
-	$(OCAMLC) -custom -o $@ -I $(LABLGTK_DIR) $(MLLIBS) $^ $(CRYPTO_LIB) $(SQLITE_LINK)
+	$(OCAMLC) -custom -o $@ -I $(LABLGTK_DIR) $(MLLIBS) $^ -cclib "$(CRYPTO_LIB)" $(SQLITE_LINK)
 monotone-viz : MLLIBS = str.cma lablgtk.cma gtkInit.cmo lablgnomecanvas.cma 
 ifeq ($(LOCAL_LABLGTK), yes)
 $(OBJ) : .lablgtk
